/** (C) Copyright 2013, Applied Physical Sciences Corp., A General Dynamics Company
 **
 ** Gravity is free software; you can redistribute it and/or modify
 ** it under the terms of the GNU Lesser General Public License as published by
 ** the Free Software Foundation; either version 3 of the License, or
 ** (at your option) any later version.
 **
 ** This program is distributed in the hope that it will be useful,
 ** but WITHOUT ANY WARRANTY; without even the implied warranty of
 ** MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 ** GNU Lesser General Public License for more details.
 **
 ** You should have received a copy of the GNU Lesser General Public
 ** License along with this program;
 ** If not, see <http://www.gnu.org/licenses/>.
 **
 */

syntax = "proto2";
option optimize_for = SPEED;
option java_outer_classname = "ComponentDataLookupResponseContainer";
option java_package = "com.aphysci.gravity.protobuf";

package gravity;

message PublisherInfoPB
{
	optional string url               = 1;
	optional bool isRelay             = 2;
	optional string componentID       = 3;
	optional string ipAddress         = 4;
	optional uint32 registration_time = 5;
}

message ComponentDataLookupResponsePB
{
	optional string lookupID            = 1;
	repeated PublisherInfoPB publishers = 2;
	optional string domain_id           = 3 [default = ""];
}
